### DISCUSSION TITLE
Trying to npm start /wave/website

### DISCUSSION THREAD

**Discussion by annahubbard** [h2oai/wave#1649] [General]
Hello! I am trying to adda button on website/src/pages/index.js but I cannot get the installation instructions going. I have made a virtual env and activated it through the anaconda prompt 
conda create -n venv
conda activate venv
Then i go to wsl terminal and try to install and start in the same dir
https://github.com/h2oai/wave/blob/master/website/README.md
git clone https://github.com/h2oai/wave.git
cd wave/py
make setup
make docs
cd ../tools/showcase
make setup
make generate
cd ../../website
make ci
but my code fails on make setup 

Checking out files: 100% (34/34), done.
python3 -m venv venv
The virtual environment was not created successfully because ensurepip is not
available.  On Debian/Ubuntu systems, you need to install the python3-venv
package using the following command.

    apt-get install python3-venv

You may need to use sudo with that command.  After installing the python3-venv
package, recreate your virtual environment.

Failing command: ['/mnt/c/users/anna/desktop/try_os_sad/wave/py/venv/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']

Makefile:17: recipe for target 'setup' failed
make: *** [setup] Error 1

**Comment by mturoci** [comment]
Hi @annahubbard.

You first need to [setup your dev env](https://wave.h2o.ai/docs/contributing/#development-setup) prior to running the `website` project.

> I have made a virtual env and activated it through the anaconda prompt

We do not use anaconda anywhere when installing from sources, everything should be automatically setup for you when running `make all` from the root of the project.

> The virtual environment was not created successfully because ensurepip is not
available

Seems like a WSL problem, quick google search yielded [this SO question](https://stackoverflow.com/questions/39539110/pyvenv-not-working-because-ensurepip-is-not-available) so I suggest starting there and making sure you are able to create a virtual env via pip as it's necessary for the whole project setup. Also make sure you meet all software prerequisites.

**Reply by annahubbard** [reply]
Thank you!
Is this the last thing to see after running make all
The command prompt is just hanging here 
![image](https://user-images.githubusercontent.com/36458121/193473607-7942cc4b-5e24-41bf-aff0-732e0749e72f.png)

**Reply by mturoci** [reply]
Some packages can take even 10-15mins to build. You may try [this tip](https://github.com/opencv/opencv-python/issues/391#issuecomment-755048450), but it should run to completion eventually.

Other quick-fix is to temporarily remove `opencv-python` from `py/examples/requirements.txt` since you will most likely not need it anyway.
